### PR TITLE
Add GitHub Actions Workflows.

### PR DESCRIPTION
**Pull request by bdice**
This PR adds GitHub Actions workflows to `raft`.

**Wait until after the 22.12 release is complete (and potentially until other repos are ready to migrate to GHA) to merge.**

### Task list

Coverage required for this PR:
- [x] C++ tests
  - [x] Two test failures on ARM? `Result/18: SWoRTests/SWoRTestD; /opt/conda/conda-bld/work/cpp/test/random/sample_without_replacement.cu:269`
- [x] Python tests
- [x] Codecov
- [x] Style checks

Future work:
- Build Metrics Report (copy from cudf when it is added)
- JUnit output annotations? https://github.com/marketplace/actions/junit-report-action
- Reduce exit code handling verbosity (use global variable with `trap` function)
- Set up `rapids-dependency-file-generator` pre-commit hook
- Docs builds

### CODE REVIEW

File: ci/check_style.sh
Comment (bdice) [inline ci/check_style.sh:18]:
```suggestion
```

File: conda/recipes/libraft/meta.yaml
Comment (ajschmidt8) [inline conda/recipes/libraft/meta.yaml:49]: should `cuda-profiler-api` be in `dependencies.yaml`? I don't see it there currently.

File: conda/recipes/pylibraft/meta.yaml
Comment (ajschmidt8) [inline conda/recipes/pylibraft/meta.yaml:7]:
```suggestion
{% set py_version = environ['CONDA_PY'] %}
```

File: conda/recipes/raft-dask/meta.yaml
Comment (ajschmidt8) [inline conda/recipes/raft-dask/meta.yaml:7]:
```suggestion
{% set py_version = environ['CONDA_PY'] %}
```

**Review by ajschmidt8 — COMMENTED** [review]
Awesome! Few code suggestions and two questions:

- Can we add a `dependencies.yaml @rapidsai/ops-codeowners` line in `CODEOWNERS`?
- Was there a verdict on including the `libcu*` libraries in `dependencies.yaml` vs. relying on them as `run` dependencies in `meta.yaml`?

File: ci/build_python.sh
Comment (ajschmidt8) [inline ci/build_python.sh:27]:
```suggestion
  --channel "${CPP_CHANNEL}" \
  --channel "${RAPIDS_CONDA_BLD_OUTPUT_DIR}" \
```

Since `raft-dask` depends on `pylibraft`, we need to add this extra channel to ensure that the `raft-dask` build uses the artifacts from the previous `pylibraft` build.

This is similar to what we did in `cudf` here: https://github.com/rapidsai/cudf/blob/72361a4a72670f460c074d00c978627159cdc60d/ci/build_python.sh#L18-L45

For reference, `RAPIDS_CONDA_BLD_OUTPUT_DIR` is defined/used in our CI images here: https://github.com/rapidsai/ci-imgs/blob/d8ada6950f26440d4d63fdaabc8bf8af9efb325d/Dockerfile#L104-L107

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
I didn't want to push this change myself because it's important to take note of.

File: ci/build_python.sh
Comment (ajschmidt8) [reply ci/build_python.sh:27]: Actually I'll commit this change, but leave this unresolved for visibility. @bdice feel free to mark it as resolved once you view it.

File: conda/recipes/libraft/meta.yaml
Comment (bdice) [reply conda/recipes/libraft/meta.yaml:49]: Yes. Treat `cuda-profiler-api` and `cuda-nvtx` and similar packages the same way as `libcu*`.

Resolved in 406212d27816ca0e5048949d68ff78a38f9a3943.

**Comment by bdice** [comment]
rerun tests

**Review by cjnolet — APPROVED** [review]
This looks great! I compared the old scripts and recipes w/ the new scripts and everything looks good to me.

When you guys are ready, I can help split the `dependencies.yml` apart for the different recipes.

**Comment by bdice** [comment]
> When you guys are ready, I can help split the `dependencies.yml` apart for the different recipes.

I think we'll be ready for that after this PR merges. Thanks again!

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
@gpucibot merge

**Pull request merged**
